app/Models/User.php

```diff
@@ -156,4 +146,73 @@ public function isRootAdmin()
     {
         return $this->root_admin === 1;
     }
+
+    /**
+     * Returns the user's daemon secret for a given server.
+     * @param  Server $server \Pterodactyl\Models\Server
+     * @return null|string
+     */
+    public function daemonToken(Server $server)
+    {
+        if ($this->id === $server->owner_id || $this->isRootAdmin()) {
+            return $server->daemonSecret;
+        }
+
+        $subuser = Subuser::where('server_id', $server->id)->where('user_id', $this->id)->first();
+
+        if (is_null($subuser)) {
+            return null;
+        }
+
+        return $subuser->daemonSecret;
+    }
+
+    /**
+     * Returns an array of all servers a user is able to access.
+     * Note: does not account for user admin status.
+     *
+     * @return array
+     */
+    public function serverAccessArray()
```

Comment:
Sometimes I only need the array of server IDs to use in custom queries, othertimes I want the collection.